orchagent/portsorch.cpp

```diff
@@ -448,191 +466,455 @@ PortsOrch::PortsOrch(DBConnector *db, DBConnector *stateDb, vector<table_name_wi
+    attr.value.objlist.count = static_cast<sai_uint32_t>(portList.size());
+    attr.value.objlist.list = portList.data();
 
     status = sai_switch_api->get_switch_attribute(gSwitchId, 1, &attr);
     if (status != SAI_STATUS_SUCCESS)
     {
         SWSS_LOG_ERROR("Failed to get port list, rv:%d", status);
-        task_process_status handle_status = handleSaiGetStatus(SAI_API_SWITCH, status);
+        auto handle_status = handleSaiGetStatus(SAI_API_SWITCH, status);
         if (handle_status != task_process_status::task_success)
         {
-            throw runtime_error("PortsOrch initialization failure");
+            SWSS_LOG_THROW("PortsOrch initialization failure");
         }
     }
 
-    /* Get port hardware lane info */
-    for (i = 0; i < m_portCount; i++)
+    // Get port hardware lane info
+    for (const auto &portId : portList)
     {
-        sai_uint32_t lanes[8] = { 0,0,0,0,0,0,0,0 };
+        std::vector<sai_uint32_t> laneList(Port::max_lanes, 0);
+
         attr.id = SAI_PORT_ATTR_HW_LANE_LIST;
-        attr.value.u32list.count = 8;
-        attr.value.u32list.list = lanes;
+        attr.value.u32list.count = static_cast<sai_uint32_t>(laneList.size());
+        attr.value.u32list.list = laneList.data();
 
-        status = sai_port_api->get_port_attribute(port_list[i], 1, &attr);
+        status = sai_port_api->get_port_attribute(portId, 1, &attr);
         if (status != SAI_STATUS_SUCCESS)
         {
-            SWSS_LOG_ERROR("Failed to get hardware lane list pid:%" PRIx64, port_list[i]);
-            task_process_status handle_status = handleSaiGetStatus(SAI_API_PORT, status);
+            SWSS_LOG_ERROR("Failed to get hardware lane list pid:%" PRIx64, portId);
+            auto handle_status = handleSaiGetStatus(SAI_API_PORT, status);
             if (handle_status != task_process_status::task_success)
             {
-                throw runtime_error("PortsOrch initialization failure");
+                SWSS_LOG_THROW("PortsOrch initialization failure");
             }
         }
 
-        set<int> tmp_lane_set;
-        for (j = 0; j < attr.value.u32list.count; j++)
+        std::set<std::uint32_t> laneSet;
+        for (sai_uint32_t i = 0; i < attr.value.u32list.count; i++)
         {
-            tmp_lane_set.insert(attr.value.u32list.list[j]);
+            laneSet.insert(attr.value.u32list.list[i]);
         }
 
-        string tmp_lane_str = "";
-        for (auto s : tmp_lane_set)
+        this->m_portListLaneMap[laneSet] = portId;
+
+        SWSS_LOG_NOTICE(
+            "Get port with lanes pid:%" PRIx64 " lanes:%s",
+            portId, swss::join(" ", laneSet.cbegin(), laneSet.cend()).c_str()
+        );
+    }
+}
+
+auto PortsOrch::getPortConfigState() const -> port_config_state_t
+{
+    return this->m_portConfigState;
+}
+
+void PortsOrch::setPortConfigState(port_config_state_t value)
+{
+    this->m_portConfigState = value;
+}
+
+bool PortsOrch::addPortBulk(const std::vector<PortConfig> &portList)
```

Comment:
Please add comment what this fn is doing

Reply:
@prgeor done